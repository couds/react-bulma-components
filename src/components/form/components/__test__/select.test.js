import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../select';

describe('Select component', () => {
  it('Should exist', () => {
    expect(Select).toMatchSnapshot();
  });
  it('Should have select classname', () => {
    const component = renderer.create(
      <Select>
        <option>1</option>
        <option>2</option>
        <option>3</option>
      </Select>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Select className="other-class this-is-a-test">
        <option>1</option>
        <option>2</option>
        <option>3</option>
      </Select>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(
      <Select style={{ width: '100%' }}>
        <option>1</option>
        <option>2</option>
        <option>3</option>
      </Select>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be large, red, disabled and multioption', () => {
    const component = renderer.create(
      <Select color="danger" size="large" multiple disabled>
        <option>1</option>
        <option>2</option>
        <option>3</option>
      </Select>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
