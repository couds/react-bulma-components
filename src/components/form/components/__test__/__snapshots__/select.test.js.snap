// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component Should be large, red, disabled and multioption 1`] = `
<div
  className="select is-large is-danger is-multiple"
  style={Object {}}
>
  <select
    disabled={true}
    multiple={true}
    name={null}
    readOnly={false}
    value=""
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select>
</div>
`;

exports[`Select component Should concat classname in props with Bulma classname 1`] = `
<div
  className="select other-class this-is-a-test"
  style={Object {}}
>
  <select
    disabled={false}
    multiple={false}
    name={null}
    readOnly={false}
    value=""
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select>
</div>
`;

exports[`Select component Should exist 1`] = `[Function]`;

exports[`Select component Should have select classname 1`] = `
<div
  className="select"
  style={Object {}}
>
  <select
    disabled={false}
    multiple={false}
    name={null}
    readOnly={false}
    value=""
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select>
</div>
`;

exports[`Select component Should use inline styles 1`] = `
<div
  className="select"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <select
    disabled={false}
    multiple={false}
    name={null}
    readOnly={false}
    value=""
  >
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select>
</div>
`;
